packages/skeleton/src/components/cell.ts

```diff
@@ -0,0 +1,196 @@
+      this.errors.push({
+        code: RedwoodErrorCode.CELL_NO_SUCCESS_EXPORT,
+        message: 'No "Success" export found but one is required',
+      })
+    }
+
+    // Check if the mandatory query export is present
+    if (!this.hasQueryExport) {
+      this.errors.push({
+        code: RedwoodErrorCode.CELL_NO_QUERY_EXPORT,
+        message: 'No "QUERY" export found but one is required',
+      })
+    } else {
+      const graphqlExport = namedExports.find((node) => {
+        if (
+          node.declaration != null &&
+          isVariableDeclaration(node.declaration) &&
+          isVariableDeclarator(node.declaration.declarations[0]) &&
+          isIdentifier(node.declaration.declarations[0].id)
+        ) {
+          return node.declaration.declarations[0].id.name === 'QUERY'
+        }
+        return false
+      }) as ExportNamedDeclaration // Note: This should not be undefined given this.hasQueryExport == true
+
+      if (
+        graphqlExport.declaration != null &&
+        isVariableDeclaration(graphqlExport.declaration) &&
+        isVariableDeclarator(graphqlExport.declaration.declarations[0]) &&
+        graphqlExport.declaration.declarations[0].init != null
+      ) {
+        const gqlVariable = graphqlExport.declaration.declarations[0].init
+        switch (gqlVariable.type) {
+          case 'TaggedTemplateExpression':
+            if (gqlVariable.quasi.start && gqlVariable.quasi.end) {
+              this.gqlQuery =
+                gqlVariable.quasi.quasis[0].value.cooked ||
+                gqlVariable.quasi.quasis[0].value.raw
+              this.gqlQueryName = getGraphQLQueryName(this.gqlQuery)
+              if (this.gqlQueryName === undefined) {
+                this.warnings.push({
+                  code: RedwoodWarningCode.CELL_NO_QUERY_OPERATION_NAME,
+                  message: 'We recommend that you name your query operation',
+                })
+              }
+            } else {
+              this.errors.push({
+                code: RedwoodErrorCode.GENERIC_PARSER_ERROR_JSTS,
+                message: `Could not extract the GraphQL query from "${gqlVariable.type}"`,
+              })
+            }
+            break
+          default:
+            this.errors.push({
+              code: RedwoodErrorCode.GENERIC_PARSER_ERROR_JSTS,
+              message: `Could not process the GraphQL query from "${gqlVariable.type}"`,
+            })
+            break
+        }
+      } else {
+        this.errors.push({
+          code: RedwoodErrorCode.GENERIC_PARSER_ERROR_JSTS,
+          message: 'Could not process the GraphQL variable',
+        })
+      }
+    }
+
+    // Determine if the cell is valid
+    this.isValid = this.hasQueryExport && this.hasSuccessExport
+  }
+}
+
+export function extractCell(filepath: string): RedwoodCell {
+  return new RedwoodCell(filepath)
+}
+
+export function extractCells(
+  project: RedwoodProject | undefined = undefined
+): RedwoodCell[] {
```

Comment:
You can simplify this a lot:

```suggestion
export function extractCells(project?: RedwoodProject): RedwoodCell[] {
```